### PR TITLE
[COMMUNITY] remove mentions of JIRA, update community docs

### PR DESCRIPTION
**Pull request by szha**
## Description ##
remove mentions of JIRA, update community docs

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented

### Changes ###
- [x] remove mentions of JIRA
- [x] update community docs (credit: many are adapted from tvm)
- [x] update github template
- [x] create RFC issue template

## Comments ##
- https://lists.apache.org/thread.html/rc2b2615fd9b43d90f35b584621e1f57c08e221061ffdef60c15c14cb%40%3Cdev.mxnet.apache.org%3E

### CODE REVIEW

File: docs/python_docs/themes/mx-theme/mxtheme/footer.html [hunk 1]
```diff
@@ -4,13 +4,12 @@
             <div class="col-4">
                 <h4 class="footer-category-title">Resources</h4>
                 <ul class="contact-list">
-                    <li><a class="u-email" href="mailto:dev@mxnet.apache.org">Dev list</a></li>
-                    <li><a class="u-email" href="mailto:user@mxnet.apache.org">User mailing list</a></li>
-                    <li><a href="https://cwiki.apache.org/confluence/display/MXNET/Apache+MXNet+Home">Developer Wiki</a></li>
-                    <li><a href="https://issues.apache.org/jira/projects/MXNET/issues">Jira Tracker</a></li>
-                    <li><a href="https://github.com/apache/incubator-mxnet/labels/Roadmap">Github Roadmap</a></li>
+                    <li><a href="https://lists.apache.org/list.html?dev@mxnet.apache.org">Dev list</a> (<a class="u-email" href="mailto:dev-subscribe@mxnet.apache.org">subscribe</a>)</li>
```
Comment (ys2843) [inline docs/python_docs/themes/mx-theme/mxtheme/footer.html:7]: This footer link here is different from main site. 
And the parentheses appear black while "subscribe" text is white. We can put parentheses inside the inner `<a/>` tag like this `<a class="u-email" href="mailto:dev-subscribe@mxnet.apache.org">(subscribe)</a>`

File: docs/static_site/src/pages/community/code_guide.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+-   Favor `const` member function when possible.
+-   Use [RAII](https://en.cppreference.com/w/cpp/language/raii) to manage resources, including smart pointers like shared_ptr and unique_ptr as well as allocating in constructors and deallocating in destructors. Avoid explicit calls to new and delete when possible. Use make_shared and make_unique instead.
+
+We use [`cpplint`](https://github.com/cpplint/cpplint) to enforce the code style. Because
+different version of `cpplint` might change by its version, it is
+recommended to use the same version of the `cpplint` as the master.
+You can also use the following command via docker.
+
+``` {.bash}
+ci/build.py -R --docker-registry mxnetci --platform ubuntu_cpu --docker-build-retries 3 --shm-size 500m /work/runtime_functions.sh sanity_cpp
```
Comment (ys2843) [inline docs/static_site/src/pages/community/code_guide.md:52]: Not super important, but if wrap this code block inside `{% highlight bash %}` and `{% endhighlight %}`, the clipboard feature will be added in the web page and user can click the button to copy, same for all languages. See [here](https://github.com/apache/incubator-mxnet/blob/master/docs/static_site/src/_includes/get_started/linux/python/cpu/pip.md) for usage.

**Comment by ys2843** [comment]
Preview link: http://ec2-34-219-134-42.us-west-2.compute.amazonaws.com/

File: docs/python_docs/themes/mx-theme/mxtheme/footer.html (hunk 1 above)
Comment (szha) [reply docs/python_docs/themes/mx-theme/mxtheme/footer.html:7]: thanks for catching it. I wasn't able to use the same reference link in python site as I did in static site, and thus made the compromise of linking to dev@ list. is there a solution with which I can link it to the same page as static site?

Comment (ys2843) [reply docs/python_docs/themes/mx-theme/mxtheme/footer.html]: I can only think of hardcoding the link, but I think this work around is ok

**Comment by ys2843** [comment]
Sorry I wasn't accurate about the clipboard trigger, wrapping code inside both ` ```bash ` and `{% highlight bash %}` work the same. But in the new docs, ` ``` {.bash} ` was used which won't trigger the clipboard.
I suggest reverting the last commit, and only update the new docs to use either ` ```bash ` or `{% highlight bash %}`, and not touching the rest of tutorials and api markdowns as the clipboard works on those pages.

**Comment by szha** [comment]
OK

**Comment by szha** [comment]
@mxnet-bot run ci [unix-cpu, windows-gpu]

**Pull request merged**